### DISCUSSION TITLE
June Personal Web - Implement Unit Test for CLI Apps using Golang and Cobra

### DISCUSSION THREAD

**Discussion by giscus** [ClavinJune/ClavinJune.github.io#8] [General]
# June Personal Web - Implement Unit Test for CLI Apps using Golang and Cobra

Implementing unit test for command-line interface (CLI) Apps using Golang and Cobra

https://clavinjune.dev/en/blogs/implement-unit-test-for-cli-apps-using-golang-and-cobra/

**Comment by skport** [comment]
Hi, Your article is very helpful to me.

I am having difficulty understanding the use of the -t (toggle) option, is -t used to turn on/off error sending for test case execution?

**Reply by clavinjune** [reply]
Hi @skport, thank you for reading the article!

Yes it is just used for making the test have more cases.

In the example, you have a flag called `toogle` that sending `ok` as a `string` if it's turned on, otherwise it'll send `not ok` as an `error`.

So there are will be 3 test cases:
- case where `toggle` is not turned on
- case where `toggle` is turned on using `-t`
- case where `toggle` is turned on using `--toggle`

**Reply by skport** [reply]
Thank you! Using this article for reference, I was able to implement the test cases in my Cobra code. :)
